src/test/compile-fail/feature-gate-no_gate_irrefutable_if_let_pattern.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// gate-test-irrefutable_let_pattern
+fn main() {
+    if let _ = 5 {}
```

Comment:
This should give some sort of error, right?

Reply:
Is there some place I can read about how to correctly annotate tests? And yes